Automattic/simplenote-ios
# Fix/download translations

### Fix
This PR fixes an issue where some entries in `Localizable.strings` for languages other than English were  not in the correct format. 
Simplenote expects the strings to be referenced by a `key`, but GlotPress uses the `original` value as key when exporting in the `strings` format (https://github.com/GlotPress/GlotPress-WP/blob/2272ba7232f2e75ad1f3f0617ed09749990acc85/gp-includes/formats/format-strings.php#L77).
This results in missing translations in the app.

To fix it, we can export the translations using the `JSON` [format](https://github.com/GlotPress/GlotPress-WP/blob/2272ba7232f2e75ad1f3f0617ed09749990acc85/gp-includes/formats/format-json.php) that exports the proper keys. 

Since the comments are not exported in JSON, the script now downloads both the versions and merges them in order to create the final `Localizable.strings`.


### Test
1. Modify some translated strings.
2. Run `./Scripts/update-translations.rb` and verify that the translations are downloaded correctly.
3. Build the app and verify that the translations are recognized correctly (As an example, `trash-noun` and `trash-verb` are not correctly recognized before this fix).
